xarray/core/variable.py

```diff
@@ -885,9 +884,7 @@ def copy(self, deep=True, data=None):
                 data = indexing.MemoryCachedArray(data.array)
 
             if deep:
-                if hasattr(data, "__array_function__") or isinstance(
-                    data, dask_array_type
-                ):
+                if hasattr(data, "__array_function__") or is_duck_dask_array(data):
```

Comment:
Likewise just `is_array_like`?

Reply:
yes i think we want the proposed `is_duck_array`